src/pressure/VCycleMG.cpp

```diff
@@ -248,171 +215,88 @@ void VCycleMG::pressure(Field &out, Field const &b, real t, bool sync) {
+            {
+                // prolongate
+                Prolongate(*field_error0_level, *field_error1_level, level, sync);
+                boundary->apply_boundary(data_error0_level, level - 1, type_e0, sync); // for m_error0 only Dirichlet BC
 
-#pragma acc data present(f_err0i, f_err1i, f_err1im, f_mg_tmpim, f_res1im, out)
-        {
-            // prolongate
-            Prolongate(f_err0i, f_err1i, i, sync);
-            boundary->apply_boundary(f_err0i.data, i - 1, type_e0, sync);  // for err0 only Dirichlet BC
-
-            // correct
-            if (i == 1) {  // use p=out on finest grid
-                // inner
-#pragma acc kernels present(f_err0i, out, d_iList[start_i:(end_i-start_i)]) async
-                {
-#pragma acc loop independent
-                    for (size_t j = start_i; j < end_i; ++j) {
-                        const size_t idx = d_iList[j];
-                        out[idx] += f_err0i[idx];
-                    }
-                }
-                // boundary
-#pragma acc kernels present(f_err0i, out, d_bList[start_b:(end_b-start_b)]) async
-                {
-#pragma acc loop independent
-                    for (size_t j = start_b; j < end_b; ++j) {
-                        const size_t idx = d_bList[j];
-                        out[idx] += f_err0i[idx];
-                    }
-                }
-                // smooth
-                Smooth(out, f_mg_tmpim, f_res1im, i - 1, sync);
-            } else {
                 // correct
-                // inner
-#pragma acc kernels present(f_err0i, f_err1im, d_iList[start_i:(end_i-start_i)]) async
-                {
-#pragma acc loop independent
-                    for (size_t j = start_i; j < end_i; ++j) {
-                        const size_t idx = d_iList[j];
-                        f_err1im[idx] += f_err0i[idx];
-                    }
-                }
-                // boundary
-#pragma acc kernels present(err0i, err1im, d_bList[start_b:(end_b-start_b)]) async
-                {
-#pragma acc loop independent
-                    for (size_t j = start_b; j < end_b; ++j) {
-                        const size_t idx = d_bList[j];
-                        f_err1im[idx] += f_err0i[idx];
-                    }
-                }
+                *field_error1_level_minus_1 += *field_error0_level;
                 // smooth
-                if (i == m_levels) {
-                    Solve(f_err1im, f_mg_tmpim, f_res1im, i - 1, sync);
+                if (level == m_levels) {
+                    Solve(*field_error1_level_minus_1, *field_mg_temporal_solution_level_minus_1, *field_residuum1_level_minus_1, level - 1, sync);
                 } else {
-                    Smooth(f_err1im, f_mg_tmpim, f_res1im, i - 1, sync);  // for err only Dirichlet BC
+                    Smooth(*field_error1_level_minus_1, *field_mg_temporal_solution_level_minus_1, *field_residuum1_level_minus_1, level - 1, sync); // for err only Dirichlet BC
```

Comment:
two spaces between code and comment are needed.
(maybe break this line and Solve?)

Reply:
b167247

Reply:
smooth is correct, only the coarsest level will be solved.

Reply:
33af824